WordPress/gutenberg
# Fix encoded characters in navigation link text

## Description
Fixes #30111. Renders `title.raw` instead of `title.rendered`.

## How has this been tested?

1. Add the navigation block.
2. Add a new menu item.
3. Select adding a new Link and begin typing the title of a page that doesn't exist. When you do so, include &.
4. Select the option to create a page draft.
5. See that the menu item shows the ampersand.
6. Select adding a new Post Link and begin typing the title of a page that doesn't exist. When you do so, include &.
7. Select the option to create a post draft.

## Screenshots
![Screen Shot 2021-03-24 at 6 59 13 AM](https://user-images.githubusercontent.com/1190420/112299748-7ad4f500-8c6e-11eb-821a-71027a6dfa0e.jpg)

## Types of changes
Bug fix

## Checklist:
